src/pages/index.js

```diff
@@ -133,17 +250,67 @@ addFormPopup.setEventListeners();
 /*                             Profile Information                            */
 /* -------------------------------------------------------------------------- */
 
-// Create the user info instance
-const userInfo = new UserInfo(selectors.profileName, selectors.profileProfession);
+// Create new user info instance
+const userInfo = new UserInfo(selectors.profileName, selectors.profileProfession, selectors.profileImage);
 
 // Create the edit form instance
 const editFormPopup = new PopupWithForm(selectors.editFormPopup, (values) => {
 
-    // Add the form's input to the profile section
-    userInfo.setUserInfo(values.name, values.profession);
+    // Render loading status
+    renderLoading(true, selectors.editFormButton);
+
+    // Update the user info in the server
+    api.updateUserinfo(values.name, values.profession)
+    .then(values => {
+
+        // Add the form's input to the profile section
+        userInfo.setUserInfo(values.name, values.profession);
+    })
+    .finally(() => {
+
+        // Restore pre-loading status
+        renderLoading(false, selectors.editFormButton);
+
+        // Close the edit form
+        editFormPopup.close();
+    })
+});
+
+/* -------------------------------------------------------------------------- */
+/*                             Update Avatar Form                             */
+/* -------------------------------------------------------------------------- */
+
+// Open the avatar popup when user clicks on the avatar's edit button
+avatarEditButton.addEventListener('click', () => {
+
+    // Create the avatar form
+    const avatarPopup = new PopupWithForm(selectors.avatarPopup, (formData) => {
```

Comment:
There should be only one instance of this class, there is no need to create a new one every time you open the popup

Reply:
Certainly! Thank you. I've fixed this.